Revisedgettingstarted (#811)
Agoric/documentation: main/guides/getting-started/README.md

If you're familar with JavaScript development tools such as `node`, `yarn`, and `git`:

```shell
go version
jq --version

> node --version # 18.16.0 or higher

Reviewer: **important**:

```suggestion
node --version # LTS version such as 18.16.0
```

"or higher" suggests we will support version 19, 21, etc. but only even numbered node major version are stable. I don't think we'll support version 19, 21, etc.
I think our policy is to support LTS (long term support) releases... though I'm not sure if we've gotten as far as making that a documented policy.

Author: I've updated the various :Node version 18.16.0 or later" strings to reflect this LTS versioning instead.

- To launch a terminal on Linux, use the **Terminal** application.
- To access WSL from Windows, visit the [WSL documentation](https://docs.microsoft.com/en-us/windows/wsl/).

## Install Go

Download Go from [go.dev/doc/install](https://go.dev/doc/install) and follow the instructions for your platform.

```shell
go version # Version 1.20.3 or higher
```

## Install Node.js

Download Node.js from [Node.js](https://nodejs.org/) and follow the instructions for your platform. We recommend installing the LTS version of node 18.

```shell
node --version # LTS version such as 18.16.0
```

> **Note:** Agoric will support all even numbered versions of Node.js. 

Reviewer: **important**:

```suggestion
**Note:** Agoric will support all long-term support (LTS) versions of Node.js. 
```

not all even numbered versions are always LTS versions. first node 18.1.0 comes out, then folks notice problems, then 18.2.0... by around 18.5.0, problem reports slow down and they put the LTS stamp on it.

Author: Ah. So odd numbered versions don't get the LTS stamp and only certain iterations of even numbered versions get the LTS stamp? Gotcha.

Author: Oh... and the text was changed to match your suggestion.

- To launch a terminal on Linux, use the **Terminal** application.
- To access WSL from Windows, visit the [WSL documentation](https://docs.microsoft.com/en-us/windows/wsl/).

> ## Install Go

Reviewer: ```suggestion
## Install Go version 1.20.3 or higher
```

ref:

https://github.com/Agoric/agoric-sdk/blob/b8d6697074b899630909440f0ea3caa519cdad44/repoconfig.sh#LL4C1-L4C22

Reviewer 2: _comment_

We need to centralize documentation on required versions in a single spot so there's only one spot to update. Adding versions into titles, headers, comments, multiple docs, etc. leads to chronically out of date documentation.

Reviewer: Where do you suggest? I can't think of a better place. For this audience, anyway. For the validator audience, we seem to put the required version info in each release. For the dapp / smart contract side of the SDK, we don't seem to do releases.

Perhaps putting it in the heading is unnecessary.

Author: Yeah, this page seems like a logical place to put all the versioning info.

I've updated the Go versioning info within the code blocks, and removed all versioning info from any headers.

Reviewer: fwiw, https://github.com/Agoric/agoric-sdk/blob/master/repoconfig.sh is a candidate single-source-of-truth. But importing from there into the docs repo seems kinda tricky. It might be worth having a unit test to check that they're in sync.

Download Go from [go.dev/doc/install](https://go.dev/doc/install) and follow the instructions for your platform.

```shell
go version
```

> ## Install jq

Reviewer 2: `jq` should not be a required piece of getting started. what requires it?

Reviewer: I agree that it _should_ not be required, but the proposed core eval docs (#799) use `make scenario2-setup` in `packages/cosmic-swingset` and that requires `jq`. As noted there, I'd like us to use `agoric start ...` instead of `make` but then we would have to more or less copy many of the commands out of the Makefile into the documentation. `scenario2-setup` does at least run in ci.

Author: Since the CoreEval docs now use the various agoric XXX commands rather than make scenario2-setup, I deleted all references to installing jq, since it's no longer required.

Download jq from [jqlang.github.io/jq/](https://jqlang.github.io/jq/) and follow the instructions for your platform.

```shell
jq --version
```

> ## Install Node.js 18.16.0 or Higher

Reviewer: ```suggestion
## Install Node.js LTS version such as 18.16.0
```

Author: This was changed to simply "Install Node.js". (The body of the section explains LTS versioning for Node.js)

Download Node.js from [Node.js](https://nodejs.org/) and follow the instructions for your platform. We recommend installing the LTS version of node 18.

```shell

> node --version # 18.16.0 or higher

Reviewer: ```suggestion
node --version # LTS version such as 18.16.0
```

Author: Changed.

yarn install
```

> **Note:** If you run into errors during install or build, make sure you have build-essential components installed. `gcc --version`.

Reviewer: **important**:

"build-essential" here is a noun, not an adjective. It's the name of a debian / Ubuntu package. So the note should perhaps say:

```suggestion
**Note:** If you run into errors during install or build, make sure you have the relevant devloper tools installed. For example, on debian or Ubuntu linux: `sudo apt-get install build-essential`.
```

By way of example, DigitalOcean, which has really good docs, discusses it this way:

[How To Compile and Install Packages From Source Using Make on a VPS \| DigitalOcean](https://www.digitalocean.com/community/tutorials/how-to-compile-and-install-packages-from-source-using-make-on-a-vps)

> To compile sources on Linux, you will need the package called "build-essential" on Debian-based systems and "Development Tools" on CentOS, as it contains the gcc/g++ compilers and libraries required to compile packages. To install this on Debian and Ubuntu run:
> 
> ```
> apt-get install build-essential
> ```
> And on CentOS run:
>
> ```
> yum groupinstall "Development Tools"
> ```

but they don't support Mac nor WSL.

Author: The text has been changed to match your suggestion.

**Note:** MacOS, Linux, and WSL are currently supported. There is currently **no support** for native Windows 10 or native Windows 11.

## Build the Cosmic Swingset Package

```shell

> cd packages/cosmic-swingset
> make

Reviewer: suggestion:

```suggestion
(cd packages/cosmic-swingset && make)
```

Doing the `cd` (and `make`) in a sub-shell means that the `cd ../..` below aren't necessary.

Author: Pretty slick. Change made.
